src/gui/EditWidgetIcons.cpp

```diff
@@ -223,16 +223,16 @@ void EditWidgetIcons::downloadFavicon()
         QUrl fallbackUrl = QUrl("https://icons.duckduckgo.com");
         fallbackUrl.setPath("/ip3/" + QUrl::toPercentEncoding(fullyQualifiedDomain) + ".ico");
 
-        m_urlsToTry.append(fallbackUrl);
-
         if (!hostIsIp) {
             QString secondLevelDomain = getSecondLevelDomain(m_url);
 
             if (fullyQualifiedDomain != secondLevelDomain) {
                 fallbackUrl.setPath("/ip3/" + QUrl::toPercentEncoding(secondLevelDomain) + ".ico");
-                m_urlsToTry.append(fallbackUrl);
+                m_urlsToTry.prepend(fallbackUrl);
             }
         }
+
+        m_urlsToTry.prepend(fallbackUrl);
```

Comment:
You cannot add it here, do not delete the prepend line above this if block. Otherwise you will never check the fully qualified domain.

Reply:
How does the fallback work? I can only find first() also in old code. Sorry was busy today, thx for the push